### PR TITLE
test: Python tests to run against a live network, and a few new tests…

### PR DESCRIPTION
**Pull request by SkidanovAlex**
…/fixes

1. Adding infra to run tests against a mocknet, and a sanity test that
uses it;
2. Fixing RPC calls to get account info after recent RPC change;
3. Adding a test that tests deploying and running gibberish contracts;
4. Moving around tests related to contract deployemnt;

### CODE REVIEW

File: pytest/tests/contracts/gibberish.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    wasm_blob = wasm_blob[:pos] + bytes([val]) + wasm_blob[pos + 1:]
+    tx = sign_deploy_contract_tx(nodes[0].signer_key, wasm_blob, 20 + iter_ * 2, hash_)
+    res = nodes[0].send_tx_and_wait(tx, 10)
+    print(res)
+
+    print("Invoking perturbed contract #%s" % iter_)
+
+    tx2 = sign_function_call_tx(nodes[0].signer_key, new_name, [], 100000000000, 100000000000, 20 + iter_ * 2 + 1, hash_)
+    # don't have any particular expectation for the call result
+    res = nodes[1].send_tx_and_wait(tx2, 10)
```
Comment (bowenwang1996) [inline pytest/tests/contracts/gibberish.py]: shouldn't we at least check that this function call returns some error?

File: pytest/lib/cluster.py [hunk 2]
```diff
@@ -362,10 +374,25 @@ def spin_up_node(config, near_root, node_dir, ordinal, boot_key, boot_addr, blac
     return node
 
 
+def connect_to_mocknet(config):
```
Comment (bowenwang1996) [inline pytest/lib/cluster.py]: what is mocknet?

Comment (SkidanovAlex) [reply pytest/lib/cluster.py]: It's an arbitrary network running that is not spun up by the test, against which a set of tests will be executed.
The actual nodes of the network are picked up from a local config

File: pytest/tests/contracts/gibberish.py (hunk 1 above)
Comment (SkidanovAlex) [reply pytest/tests/contracts/gibberish.py]: You cannot expect it to return an error, a valid contract with one byte changed can actually be a valid contract.

What we are actually testing is that the deployment and invoking of such a contract doesn't trigger the server going down.

File: pytest/lib/cluster.py [hunk 3]
```diff
@@ -123,7 +124,18 @@ def get_validators(self):
...
 
     def get_account(self, acc, finality='optimistic'):
-        return self.json_rpc('query', {"request_type": "view_account", "account_id": acc, "block_id": None, "finality": finality})
+        return self.json_rpc('query', {"request_type": "view_account", "account_id": acc, "finality": finality})
+
+    def get_access_key_list(self, acc, finality='optimistic'):
+        return self.json_rpc('query', {"request_type": "view_access_key_list", "account_id": acc, "finality": finality})
+
+    def get_nonce_for_pk(self, acc, pk, finality='optimistic'):
+        for access_key in self.get_access_key_list(acc, finality)['result']['keys']:
```
Comment (bowenwang1996) [inline pytest/lib/cluster.py]: this is unnecessary. You can query an access key directly with `"request_type": "view_access_key"`

File: pytest/tests/mocknet/sanity.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+import sys, time, random, base58
+
+sys.path.append('lib')
+
+from cluster import connect_to_mocknet
+from transaction import sign_payment_tx, sign_function_call_tx, sign_deploy_contract_tx
+from utils import load_binary_file
+
+nodes, accounts = connect_to_mocknet(None)
```
Comment (ailisp) [inline pytest/tests/mocknet/sanity.py]: A possibly problem here: RpcNode doesn't give you node keys, Did you init keypair else where to sign txn below?

Comment (SkidanovAlex) [reply pytest/tests/mocknet/sanity.py]: Accounts are stored separately from the nodes.
I think generally the idea of coupling nodes and account was weird, I don't want to rewrite the regular infra at this point, but for the new tests I think those should be decoupled.

**Pull request merged**